### ISSUE TITLE
Remove support for legacy logging configuration from the CO code

### ISSUE THREAD

**Issue by scholzj** [strimzi/strimzi-kafka-operator#4629]
In `v1beta2`, the only way for configuring the external logging should  be using the `configMapKeyRef` field. I.e. this is not supported anymore:

```yaml
logging:
  type: external
  name: my-config-map
```

and only this can be used instead:

```yaml
logging:
  type: external
  valueFrom:
    configMapKeyRef:
      name: my-config-map
      key: log4j.properties
```

We should remove the support for the old legacy way from the CO. This should cover all affected resources and classes.

_(The fields should still remain present in the `api` module for the time being)_

**Comment by ShubhamRwt** [comment]
@scholzj - Hello. I wanted to work on this issue

**Comment by ShubhamRwt** [comment]
For the above issue the class that I should seek should be `AbstractModel` class right?

**Comment by scholzj** [comment]
Sure, why not. @sknot-rh did originally write this code - so if you need anything he might be the best help.

**Comment by ShubhamRwt** [comment]
```(The fields should still remain present in the api module for the time being)``` @scholzj , does this mean that I don't have to make any changes in the api package?

**Comment by scholzj** [comment]
Yes, that is correct

**Comment by sknot-rh** [comment]
Should `io.strimzi.api.kafka.model.ExternalLogging#name` be removed?

**Comment by scholzj** [comment]
No, we are not touching the API as said above.

**Comment by sknot-rh** [comment]
Yeah, I am just double checking. Thanks.

**Comment by scholzj** [comment]
Closed in #4717
